### PR TITLE
Bundle name collisions

### LINKED ISSUE #1303: Services with the same name clash when deploying a bundle
If you have wordpress on your canvas, then you add a bundle which includes wordpress then the two services clash and it puts the model into a weird state where they are both shown on the canvas but only one exists in the added services list.

### PR DESCRIPTION
**Pull request by huwshimi**
Handle adding a bundle with a service name that matches an existing service. Fixes #1303.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add wordpress
- add the wordpress bundle
- you should now have two wordpress services: "wordpress" and "wordpress-a" and there should be no errors in your browser console.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4314/
Test PASSed.

**Comment by anthonydillon** [comment]
Code looks :ok: QA issue :+1: but spotted this issue (https://github.com/juju/juju-gui/issues/1311) but outside the scope of this branch.

File: jujugui/static/gui/src/app/views/utils.js [hunk 1]
```diff
@@ -1810,6 +1810,57 @@ YUI.add('juju-view-utils', function(Y) {
...
+      var serviceCharmId = utils._extractCharmName(service.get('charm'));
+      var providedCharmId = utils._extractCharmName(charmId);
+      // If we have a matching charm then increase the count.
+      if (serviceCharmId === providedCharmId) {
+        charmCount += 1;
+      }
+    });
+    // In some cases the service has already been added to the db so we need to
+    // ignore it in the final count.
+    if (ignoreSelf) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: I don't really understand this - it's supposed to be generating the name to be added to the db...so how could it be in the db already?

**Comment by hatched** [comment]
:+1: :ok: With one question that might be a 'fix' for something that shouldn't happen.

File: jujugui/static/gui/src/app/views/utils.js (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/views/utils.js]: It creates a reference in the db when it does `this.db.services.ghostService(charm);` above. See this line: https://github.com/juju/juju-gui/pull/1310/files#diff-1dcaa888078db4ae76c5fb071e2ead75R432

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**